### PR TITLE
Fix savewhen issues

### PR DESCRIPTION
**Pull request by JoranAngevaare**
Fix some issues introduced in features added in https://github.com/AxFoundation/strax/pull/631

also see https://github.com/XENONnT/straxen/pull/879

### CODE REVIEW

**Comment by WenzDaniel** [comment]
Hmm I thought we covered everything with the tests added in 631. What have we overlooked?

**Comment by JoranAngevaare** [comment]
Some functions that use the lines I changed here and are failing in https://github.com/XENONnT/straxen/pull/879

File: strax/run_selection.py [hunk 1]
```diff
@@ -424,7 +424,8 @@ def available_for_run(self: strax.Context,
     is_stored = defaultdict(list)
     for target in self._plugin_class_registry.keys():
         # Skip targets that are not stored
-        if not self._plugin_class_registry[target].save_when > strax.SaveWhen.NEVER:
+        save_when = self.get_save_when(target)
+        if save_when == strax.SaveWhen.NEVER:
```
Comment (JoranAngevaare) [inline strax/run_selection.py:428]: issue #3

**Review by WenzDaniel — APPROVED** [review]
Thanks for fixing. Looks good.

**Pull request merged**